### ISSUE TITLE
TechDocs: `EntityTechdocsContent` not rendering sub-pages

### ISSUE THREAD

**Issue by martinohmann** [backstage/backstage#11410]
## Expected Behavior

When visiting an existing sub-page via the docs tab, it should render the techdocs content.

For example, the following urls should both render the techdocs for `docs/subpage.md`:

- TechDocs plugin: `https://backstage-host.tld/docs/default/Component/my-component/subpage/`
- Catalog Entity docs tab: `https://backstage-host.tld/catalog/default/component/my-component/docs/subpage/`

## Actual Behavior

When visiting the subpage via the TechDocs plugin (`https://backstage-host.tld/docs/default/Component/my-component/subpage/`), it renders correctly.

When visiting it via the Catalog Entity docs tab (`https://backstage-host.tld/catalog/default/component/my-component/docs/subpage/`), it just renders an empty `EntityTechdocsContent`. There are no errors in the JavaScript Console, so I suspect that this might be a routing issue.

**Please note that the docs index (`docs/index.md`) renders correctly using the docs tab, but this seems to be the only page that does.**

## Steps to Reproduce

1. Add the docs tab to the entity page:

```tsx
const defaultEntityPage = (
  <EntityLayoutWrapper>
    <EntityLayout.Route path="/docs" title="Docs">
      <EntityTechdocsContent />
    </EntityLayout.Route>
  </EntityLayoutWrapper>
);
```

2. Visit a subpage of the techdocs, once via the docs tab and once via the "View techdocs" link on the entity page.

3. The subpage visited via the docs tab renders an empty `EntityTechdocsContent`, while it correctly shows the documentation when navigated via the "View techdocs" link.

## Your Environment

- Browser Information: 

- Output of `yarn backstage-cli info`: 

```text
OS:   Darwin 20.6.0 - darwin/arm64
node: v16.15.0
yarn: 1.22.18
cli:  0.17.1-next.1 (installed)

Dependencies:
  @backstage/app-defaults                               1.0.2-next.0
  @backstage/backend-common                             0.13.3-next.1
  @backstage/backend-tasks                              0.3.1-next.0
  @backstage/catalog-client                             1.0.1
  @backstage/catalog-model                              1.0.1
  @backstage/cli-common                                 0.1.8
  @backstage/cli                                        0.17.1-next.1
  @backstage/config-loader                              1.1.1-next.0
  @backstage/config                                     1.0.0
  @backstage/core-app-api                               1.0.2-next.0
  @backstage/core-components                            0.9.4-next.0
  @backstage/core-plugin-api                            1.0.2-next.0
  @backstage/errors                                     1.0.0
  @backstage/integration-react                          1.1.0-next.1
  @backstage/integration                                1.2.0-next.0
  @backstage/plugin-adr-backend                         0.1.0-next.0
  @backstage/plugin-adr-common                          0.1.0-next.0
  @backstage/plugin-adr                                 0.1.0-next.0
  @backstage/plugin-api-docs                            0.8.5-next.1
  @backstage/plugin-app-backend                         0.3.32-next.0
  @backstage/plugin-auth-backend                        0.13.1-next.1
  @backstage/plugin-auth-node                           0.2.1-next.0
  @backstage/plugin-badges-backend                      0.1.26-next.0
  @backstage/plugin-badges                              0.2.29-next.1
  @backstage/plugin-catalog-backend-module-aws          0.1.5-next.0
  @backstage/plugin-catalog-backend-module-bitbucket    0.1.3-next.0
  @backstage/plugin-catalog-backend-module-gerrit       0.1.0-next.1
  @backstage/plugin-catalog-backend-module-github       0.1.3-next.0
  @backstage/plugin-catalog-backend-module-msgraph      0.3.2-next.0
  @backstage/plugin-catalog-backend                     1.1.2-next.1
  @backstage/plugin-catalog-common                      1.0.1
  @backstage/plugin-catalog-graph                       0.2.17-next.1
  @backstage/plugin-catalog-import                      0.8.8-next.1
  @backstage/plugin-catalog-react                       1.1.0-next.1
  @backstage/plugin-catalog                             1.2.0-next.1
  @backstage/plugin-explore-react                       0.0.17-next.0
  @backstage/plugin-explore                             0.3.36-next.1
  @backstage/plugin-github-actions                      0.5.5-next.1
  @backstage/plugin-org                                 0.5.5-next.1
  @backstage/plugin-permission-common                   0.6.0
  @backstage/plugin-permission-node                     0.6.1-next.0
  @backstage/plugin-permission-react                    0.4.1-next.0
  @backstage/plugin-proxy-backend                       0.2.26-next.0
  @backstage/plugin-scaffolder-backend                  1.2.0-next.0
  @backstage/plugin-scaffolder-common                   1.0.1
  @backstage/plugin-scaffolder                          1.2.0-next.1
  @backstage/plugin-search-backend-node                 0.6.1-next.0
  @backstage/plugin-search-backend                      0.5.2-next.0
  @backstage/plugin-search-common                       0.3.3
  @backstage/plugin-search-react                        0.2.0-next.1
  @backstage/plugin-search                              0.8.1-next.1
  @backstage/plugin-tech-insights-backend-module-jsonfc 0.1.16-next.1
  @backstage/plugin-tech-insights-backend               0.4.0-next.1
  @backstage/plugin-tech-insights-common                0.2.4
  @backstage/plugin-tech-insights-node                  0.3.0-next.1
  @backstage/plugin-tech-insights                       0.2.1-next.1
  @backstage/plugin-tech-radar                          0.5.12-next.0
  @backstage/plugin-techdocs-backend                    1.1.1-next.0
  @backstage/plugin-techdocs-node                       1.1.1-next.0
  @backstage/plugin-techdocs-react                      0.1.1-next.1
  @backstage/plugin-techdocs                            1.1.1-next.1
  @backstage/plugin-user-settings                       0.4.4-next.0
  @backstage/release-manifests                          0.0.2
  @backstage/search-common                              0.3.3
  @backstage/test-utils                                 1.1.0-next.1
  @backstage/theme                                      0.2.15
  @backstage/types                                      1.0.0
  @backstage/version-bridge                             1.0.1
```

**Comment by garyniemen** [comment]
@camilaibs can you look into this one

**Comment by camilaibs** [comment]
Hi @martinohmann, thanks for reporting, we appreciate it!
This issue appears to be the same one reported [here](https://github.com/backstage/backstage/issues/11536) and a  [fix](https://github.com/backstage/backstage/pull/11542) was merged yesterday so it should be available in [Backstage v1.2.0](https://github.com/backstage/backstage/releases/tag/v1.2.0).
Could you update and see if it's resolved for you?
